src/components/Articles.jsx

```diff
@@ -9,57 +9,136 @@ export default function Articles() {
+        let sortCriterion;
+
+        switch (selectedItem.split(" ")[0]) {
+            case "Date":
+                sortCriterion = "created_at";
+                break;
+            case "Title":
+                sortCriterion = "title";
+                break;
+            case "Author":
+                sortCriterion = "author";
+                break;
+            case "Comments":
+                sortCriterion = "comment_count";
+                break;
+            default:
+                break;
+        }
+
+        setSortParams({
+            sort: sortCriterion,
+            order: sortOrder
+        });
+    };
+
     return (
-        <ul className="Articles">
-            {!isLoading ? (
-                articles.map((article) => {
-                    let photoSrc, photoAlt;
-                    switch (article.topic) {
-                        case "cooking":
-                            photoSrc = cookingPhoto;
-                            photoAlt = "colourful cookery";
-                            break;
-                        case "coding":
-                            photoSrc = codingPhoto;
-                            photoAlt = "messy computer code";
-                            break;
-                        case "football":
-                        // if anything goes awry, the default will render a football photo
-                        default:
-                            photoSrc = footballPhoto;
-                            photoAlt = "football stadium";
-                            break;
-                    }
-                    return (
-                        <li key={article.article_id}>
-                            <Link to={`/articles/${article.article_id}`}>
-                                <img src={photoSrc} alt={photoAlt} />
-                                <br />
-                                <strong>{article.title}</strong> <i>by</i>{" "}
-                                {article.author}
-                                <br />
-                                <span className="Articles--topic">
-                                    #{article.topic}
-                                </span>
-                            </Link>
-                        </li>
-                    );
-                })
-            ) : (
-                <p>Loading articles...</p>
-            )}
-        </ul>
+        <main className="Articles">
+            <select onChange={handleSort} ref={ref}>
+                <option>Sort articles by...</option>
+                <option>Date (ascending)</option>
+                <option>Date (descending)</option>
+                <option>Title (ascending)</option>
+                <option>Title (descending)</option>
+                <option>Author (ascending)</option>
+                <option>Author (descending)</option>
+                <option>Comments (ascending)</option>
+                <option>Comments (descending)</option>
+            </select>
```

Comment:
I wonder if adding a value to your options will allow you to refactor some of the if else statements. You could use the value to store the required naming for use of your API whilst still rendering user friendly data

Reply:
Excellent advice! Done.